cub/benchmarks/bench/transform/babelstream.cu

```diff
@@ -0,0 +1,237 @@
+// SPDX-FileCopyrightText: Copyright (c) 2024, NVIDIA CORPORATION. All rights reserved.
+// SPDX-License-Identifier: BSD-3-Clause
+
+// %RANGE% TUNE_THREADS tpb 128:1024:128
+// %RANGE% TUNE_ITEMS_PER_THREAD ipt 1:24:1
+// %RANGE% TUNE_ALGORITHM alg 0:4:1
+
+// keep checks at the top so compilation of discarded variants fails really fast
+#if !TUNE_BASE
+#  if TUNE_ALGORITHM == 4 && __CUDA_ARCH_LIST__ < 900
+#    error "Cannot compile algorithm 4 (ublkcp) below sm90"
+#  endif
+
+#  if TUNE_ALGORITHM != 2 && TUNE_ITEMS_PER_THREAD != 1
+#    error "Skip ITEMS_PER_THREAD tuning for algorithms other than 2"
+#  endif
+#endif
```

Comment:
@gevtushenko even though I have those checks at the top of the file, `search.py` still takes 2s before it skips the benchmark:
```
2024-08-19 17:18:37,668: starting build for cub.bench.transform.babelstream.tpb_896.ipt_22.alg_0: cmake --build . --target cub.bench.transform.babelstream.variant
2024-08-19 17:18:39,791: finished build for cub.bench.transform.babelstream.tpb_896.ipt_22.alg_0 (exit code: 1) in 2.122s
2024-08-19 17:18:39,818: found cached base build for cub.bench.transform.babelstream.base
2024-08-19 17:18:39,825: found cached base build for cub.bench.transform.babelstream.base
2024-08-19 17:18:39,828: starting build for cub.bench.transform.babelstream.tpb_640.ipt_24.alg_3: cmake --build . --target cub.bench.transform.babelstream.variant
2024-08-19 17:18:42,000: finished build for cub.bench.transform.babelstream.tpb_640.ipt_24.alg_3 (exit code: 1) in 2.172s
2024-08-19 17:18:42,027: found cached base build for cub.bench.transform.babelstream.base
2024-08-19 17:18:42,034: found cached base build for cub.bench.transform.babelstream.base
2024-08-19 17:18:42,036: starting build for cub.bench.transform.babelstream.tpb_384.ipt_5.alg_0: cmake --build . --target cub.bench.transform.babelstream.variant
2024-08-19 17:18:44,109: finished build for cub.bench.transform.babelstream.tpb_384.ipt_5.alg_0 (exit code: 1) in 2.072s
```
Is there a faster way to skip combinations?

Reply:
It's an effect of the network filesystem.